src/lib/database.ts

```diff
@@ -0,0 +1,131 @@
+import fs from 'node:fs';
+import readline from 'node:readline';
+import { Transform, TransformCallback } from 'node:stream';
+import zlib from 'node:zlib';
+
+import { createExternalizedPromise } from './promise';
+
+export enum SqlDumpType {
+	MYDUMPER = 'MYDUMPER',
+	MYSQLDUMP = 'MYSQLDUMP',
+}
+
+export interface SqlDumpDetails {
+	type: SqlDumpType;
+	sourceDb: string;
+}
+
+export const getSqlDumpDetails = async ( filePath: string ): Promise< SqlDumpDetails > => {
```

Comment:
This function checks if this sql dump is a mydumper dump, a mysqldump, and the source DB, if it's a mydumper dump.